src/app/core/services/asset-property-mapper.ts

```diff
@@ -18,149 +10,111 @@ import {Asset} from './models/asset';
+    };
   }
 
-  buildProperties(
-    formValue: AssetEditorDialogFormValue,
-  ): Record<string, string> {
-    const {metadata, advanced, datasource} = formValue;
-    const props: Record<string, string | null> = {};
-    props[AssetProperties.id] = trimmedOrNull(metadata?.id);
-    props[AssetProperties.name] = trimmedOrNull(metadata?.name);
-    props[AssetProperties.version] = trimmedOrNull(metadata?.version);
-    props[AssetProperties.curatorOrganizationName] = trimmedOrNull(
-      this.config.curatorOrganizationName,
-    );
-    props[AssetProperties.keywords] = trimmedOrNull(
-      metadata?.keywords?.join(', '),
-    );
-    props[AssetProperties.contentType] = trimmedOrNull(metadata?.contentType);
-    props[AssetProperties.description] = trimmedOrNull(metadata?.description);
-    props[AssetProperties.language] = metadata?.language?.id ?? null;
+  convertToAdditionalProperties(asset: UiAsset): AdditionalAssetProperty[] {
+    let result: AdditionalAssetProperty[] = [];
+    type AssetKey =
+      | 'additionalProperties'
+      | 'additionalJsonProperties'
+      | 'privateProperties'
+      | 'privateJsonProperties';
 
-    props[AssetProperties.publisher] = trimmedOrNull(metadata?.publisher);
-    props[AssetProperties.standardLicense] = trimmedOrNull(
-      metadata?.standardLicense,
-    );
-    props[AssetProperties.endpointDocumentation] = trimmedOrNull(
-      metadata?.endpointDocumentation,
-    );
+    const propertiesToConvert: AssetKey[] = [
+      'additionalProperties',
+      'additionalJsonProperties',
+      'privateProperties',
+      'privateJsonProperties',
+    ];
 
-    if (this.activeFeatureSet.hasMdsFields()) {
-      props[AssetProperties.dataCategory] = advanced?.dataCategory?.id ?? null;
-      props[AssetProperties.dataSubcategory] =
-        advanced?.dataSubcategory?.id ?? null;
-      props[AssetProperties.dataModel] = trimmedOrNull(advanced?.dataModel);
-      props[AssetProperties.geoReferenceMethod] = trimmedOrNull(
-        advanced?.geoReferenceMethod,
-      );
-      props[AssetProperties.transportMode] =
-        advanced?.transportMode?.id ?? null;
+    for (let propName of propertiesToConvert) {
+      const propValue = asset[propName];
+      if (propValue) {
+        for (let key in propValue) {
+          result.push({key: key, value: propValue[key]});
+        }
+      }
     }
+    return result;
+  }
 
-    if (datasource?.dataAddressType === 'Http') {
-      props[AssetProperties.httpProxyMethod] = this._encodeBoolean(
-        datasource?.httpProxyMethod,
-      );
-      props[AssetProperties.httpProxyPath] = this._encodeBoolean(
-        datasource?.httpProxyPath,
-      );
-      props[AssetProperties.httpProxyQueryParams] = this._encodeBoolean(
-        datasource?.httpProxyQueryParams,
-      );
-      props[AssetProperties.httpProxyBody] = this._encodeBoolean(
-        datasource?.httpProxyBody,
-      );
+  private buildAdditionalPropertiesforUiAsset(
```

Comment:
very cool, but we should remove this already, no need to implement a "Record<string, string> -> UiAsset" for now, we'll have the broker running on the UiAsset rather sooner than later